### PR TITLE
Allow easier setting of background color for TextEdit

### PR DESCRIPTION
**Pull request by bircni**
* Closes <https://github.com/emilk/egui/issues/5183>
* [x] I have followed the instructions in the PR template

### CODE REVIEW

**Comment by lucasmerlin** [comment]
Is there a reason why one would often need to change the background color of TextEdits?
Someone could argue that if we have this functionality for the background, we should also add functions to customize the font, size, padding, etc... I think this could get out of hand quickly.
If something like this is desirable be useful I think we should come up with some way to generalize this and add this to all widgets (and I think then this is related to #3284)

**Comment by AlexanderSchuetz97** [comment]
@lucasmerlin 
I require this for 2 reasons:
Really highlight text edits in red if they contain invaluid data that the human must change.

I use gray background for every other textbox because with white background users didnt even realize the textbox exists until they moved their cursor over them.

File: crates/egui/src/widgets/text_edit/builder.rs [hunk 1]
```diff
@@ -201,6 +203,13 @@ impl<'t> TextEdit<'t> {
         self
     }
 
+    /// Set the background color of the [`TextEdit`].
```
Comment (lucasmerlin) [inline crates/egui/src/widgets/text_edit/builder.rs:206]: Maybe you can add a comment that by default this comes from `Visuals::extreme_bg_color`?

**Review by lucasmerlin — APPROVED** [review]
I talked with @emilk and we decided that since extreme_bg_color is kind of obscure it makes sense to add this until #3284 is implemented

**Comment by bircni** [comment]
Okay! I will also add a comment to refer to the issue you mentioned

**Comment by lucasmerlin** [comment]
Thank you!

**Pull request merged**